pki/deploy.go

```diff
@@ -119,3 +122,197 @@ func RemoveAdminConfig(ctx context.Context, localConfigPath string) {
 	}
 	log.Infof(ctx, "Local admin Kubeconfig removed successfully")
 }
+
+func DeployCertificatesOnHost(ctx context.Context, host *hosts.Host, crtMap map[string]CertificatePKI, certDownloaderImage string) error {
+	crtList := []string{
+		CACertName,
+		KubeAPICertName,
+		KubeControllerName,
+		KubeSchedulerName,
+		KubeProxyName,
+		KubeNodeName,
+		KubeAdminCommonName,
+	}
+
+	env := []string{
+		"CRTS_DEPLOY_PATH=" + TempCertPath,
+	}
+	for _, crtName := range crtList {
+		c := crtMap[crtName]
+		// We don't need to edit the cert paths, they are not used in deployment
+		env = append(env, c.ToEnv()...)
+	}
+	return doRunDeployer(ctx, host, env, certDownloaderImage)
+}
+
+func FetchCertificatesFromHost(ctx context.Context, host *hosts.Host, image, localConfigPath string) (map[string]CertificatePKI, error) {
+	// rebuilding the certificates. This should look better after refactoring pki
+
+	tmpCerts := make(map[string]CertificatePKI)
+
+	certEnvMap := map[string][]string{
+		CACertName:          []string{CACertPath, CAKeyPath},
+		KubeAPICertName:     []string{KubeAPICertPath, KubeAPIKeyPath},
+		KubeControllerName:  []string{KubeControllerCertPath, KubeControllerKeyPath, KubeControllerConfigPath},
+		KubeSchedulerName:   []string{KubeSchedulerCertPath, KubeSchedulerKeyPath, KubeSchedulerConfigPath},
+		KubeProxyName:       []string{KubeProxyCertPath, KubeProxyKeyPath, KubeProxyConfigPath},
+		KubeNodeName:        []string{KubeNodeCertPath, KubeNodeKeyPath, KubeNodeConfigPath},
+		KubeAdminCommonName: []string{"kube-admin.pem", "kube-admin-key.pem"},
+	}
+	// get files from hosts
+
+	for crtName, certEnv := range certEnvMap {
+		certificate := CertificatePKI{}
+		crt, err := fetchFileFromHost(ctx, getTempPath(certEnv[0]), image, host)
+		if err != nil {
+			return nil, err
+		}
+		key, err := fetchFileFromHost(ctx, getTempPath(certEnv[1]), image, host)
+
+		if len(certEnv) > 2 {
+			config, err := fetchFileFromHost(ctx, getTempPath(certEnv[2]), image, host)
+			if err != nil {
+				return nil, err
+			}
+			certificate.Config = config
+		}
+		parsedCert, _ := cert.ParseCertsPEM([]byte(crt))
```

Comment:
can we add check for errors here, the certificate maybe returned with invalid data